packages/protocol/scripts/bash/revoke_contracts.sh

```diff
@@ -0,0 +1,68 @@
+#! /usr/bin/env bash
+set -euo pipefail
+rejects=()
+
+if [ "$#" -ne 1 ]; then
+  echo "Expected usage: ./revoke_contracts.sh contracts.txt"
+  echo "contracts.txt should include contracts you'd like to revoke and refund, one address per line"
+  exit 1
+fi
+
+for contract in $(<"$1"); do
+  echo "Contract $contract"
+
+  # Check that account has been created
+  accountExists=1
+  celocli lockedgold:show "$contract" > /dev/null 2>&1 || accountExists=0
+  if [ $accountExists -eq 0 ]; then
+    echo "Account not created, rejecting $contract"
+    rejects+=( $contract )
+    continue
+  fi
+
+  # Check that CELO has been unlocked
+  lockedgold=$(celocli lockedgold:show "$contract")
+  total=$(echo "$lockedgold" | grep -o "total: [0-9]*" | cut -f2 -d : | tr -d '[:space:]')
+  if [ "$total" != "0" ]; then
+    echo "Reject $contract"
+    rejects+=( $contract )
+    continue
+  fi
+
+  # Check that withdrawals are available
+  if [ -z $(echo "$lockedgold" | grep -o "time: [0-9]*" | cut -f2 -d :) ]; then
+    echo "No pending withdrawals"
+  else 
+    pendingTimes=$(echo "$lockedgold" | grep -o "time: [0-9]*" | cut -f2 -d :)
+    now=$(date +%s)
+    while IFS= read -r line
+    do 
+      if [ "$line" -ge "$now" ]; then
+        echo "Pending withdrawal not ready, rejecting $contract"
+        rejects+=( $contract )
+        continue 2
+      fi
+    done < <(printf '%s\n' "$pendingTimes")
+  fi
+
+  contractExists=1
+  celocli releasegold:revoke --contract "$contract" --yesreally > /dev/null || contractExists=0
+  if [ "$contractExists" -eq 0 ]; then
+    echo "Contract doesn't exist, rejecting $contract"
+    rejects+=( $contract )
+    continue
+  fi
+  celocli releasegold:locked-gold --contract "$contract" --action withdraw --value "$total" --yes 
```

Comment:
What happens here if there are no pending withdrawals?

Reply:
It's a no-op if no pending withdrawals so this should be ok